### PR TITLE
#2260 work package templates use all work pack templates hook

### LINKED ISSUE #2260: Work Package Templates - Use All Work Pack Templates Hook
### Description

Blocked By: #2156 #2148

We need a way to access all of the work package templates in our database on Finishline. This ticket is to make a frontend hook that retrieves all work package templates using the get all work packages endpoint.

### Acceptance Criteria

 - [ ] Create a hook to retrieve all work package templates
 - [ ] Connect it to the backend endpoint via an api call

### Proposed Solution

Look at the other get all hooks in the codebase specifically in the "hooks" folder. To find an example in vscode press ctrl + p to open the search on the top and then type projects.hooks.ts and it should appear.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Jake-Hensley**
## Changes

Created the hook, api, and url for the work package templates.

## To Do

- [ ] create the endpoint

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2260

### CODE REVIEW

**Review by bderbs30 — DISMISSED** [review]
LGTM 💸

File: src/frontend/src/utils/urls.ts [hunk 1]
```diff
@@ -135,11 +134,15 @@ const bomCreateUnit = () => `${bomEndpoints()}/units/create`;
 /************** Design Review Endpoints *******************************/
 const designReviews = () => `${API_URL}/design-reviews`;
 const designReviewsCreate = () => `${designReviews()}/create`;
+const teamTypes = () => `${designReviews()}/teamType/all`;
```
Comment (walker-sean) [inline src/frontend/src/utils/urls.ts:137]: This does fit with design reviews, but we have the corresponding route in the teams file so it probably makes more sense to keep it where it is imo
Why was this moved ?

Comment (bderbs30) [reply src/frontend/src/utils/urls.ts:137]: I’m pretty sure this was coming from the merge with develop

Comment (RChandler234) [reply src/frontend/src/utils/urls.ts:137]: agree with Sean, it should defo be with the teams stuff

File: src/frontend/src/utils/urls.ts [hunk 2]
```diff
@@ -145,6 +145,9 @@ const designReviewById = (id: string) => `${designReviews()}/${id}`;
 const designReviewDelete = (id: string) => `${designReviewById(id)}/delete`;
 const designReviewMarkUserConfirmed = (id: string) => `${designReviewById(id)}/confirm-schedule`;
 
+/**************** Work Package Template Endpoints ****************/
+const workPackageTemplates = () => `${API_URL}/templates`;
```
Comment (Peyton-McKee) [inline src/frontend/src/utils/urls.ts:149]: this doesnt match up with what aaryan used for the template endpoints. The backend endpoint is on the workpackages router which means its off of /work-packages or some shit

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]

File: src/frontend/src/utils/urls.ts [hunk 3]
```diff
@@ -85,7 +85,7 @@ const teamsSetHead = (id: string) => `${teamsById(id)}/set-head`;
 const teamsSetDescription = (id: string) => `${teamsById(id)}/edit-description`;
 const teamsCreate = () => `${teams()}/create`;
 const teamsSetLeads = (id: string) => `${teamsById(id)}/set-leads`;
-const teamTypes = () => `${teams()}/teamType/all`;
+const teamTypes = () => `${designReviews()}/teamType/all`;
```
Comment (Peyton-McKee) [inline src/frontend/src/utils/urls.ts:88]: why did you change this

File: src/frontend/src/utils/urls.ts [hunk 4]
```diff
@@ -148,6 +148,9 @@ const designReviewById = (id: string) => `${designReviews()}/${id}`;
 const designReviewDelete = (id: string) => `${designReviewById(id)}/delete`;
 const designReviewMarkUserConfirmed = (id: string) => `${designReviewById(id)}/confirm-schedule`;
 
+/**************** Work Package Template Endpoints ****************/
+const workPackageTemplates = () => `${API_URL}/work-packages/template`;
```
Comment (Peyton-McKee) [inline src/frontend/src/utils/urls.ts:152]: use the work_packages endpoint that already exists and can you move down the other work package template routes that are in the work package section down here

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]

File: src/frontend/src/utils/urls.ts (hunk 3 above)
Comment (bderbs30) [reply src/frontend/src/utils/urls.ts:88]: no clue...

File: src/frontend/src/utils/urls.ts [hunk 5]
```diff
@@ -148,6 +146,11 @@ const designReviewById = (id: string) => `${designReviews()}/${id}`;
 const designReviewDelete = (id: string) => `${designReviewById(id)}/delete`;
 const designReviewMarkUserConfirmed = (id: string) => `${designReviewById(id)}/confirm-schedule`;
 
+/******************* Work Package Template Endpoints********************/
+
+const workPackageTemplates = () => `${workPackages()}/template`;
```
Comment (Peyton-McKee) [inline src/frontend/src/utils/urls.ts:151]: dont forget to change this one

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
